app/controllers/omniauth_controller.rb

```diff
@@ -14,19 +14,25 @@ def tra_openid_connect
 
     # @user.persisted? checks that it exists and has been persisted to the database
     # @user.save checks that any changes made to an existing record have been persisted to that persisted record
-    if @user.persisted? && @user.save
+    if @user.persisted? && (user_saved = @user.save)
       session["user_id"] = @user.id
       @user.set_closed_registration_feature_flag
       sign_in_and_redirect @user
     else
+      # TODO: need a feature test for the error scenario:
+      # - uid has already been taken
+      user_id_with_clashing_email = User.find_by(email: provider_data.info.email).id
```

Comment:
defensive _but_ this is already doing something we don't understand so better to use safe navigation operator `&.id`